index.js

```diff
@@ -151,7 +151,11 @@ class Transaction {
     if (chainId < 0) chainId = 0
 
     // set chainId
-    this._chainId = chainId || data.chainId || 0
+    if (opts.chain || opts.common) {
+      this._chainId = this._common.chainId()
+    } else {
+      this._chainId = chainId || data.chainId || 0
```

Comment:
The discussion on [here](https://github.com/ethereumjs/ethereumjs-tx/pull/131#issuecomment-478304415) is pretty useful - can we add an inline comment to summarize (or paste the link to comment)? This would help future devs to gather context quickly.